Kernel/API/Jail.h

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/EnumBits.h>
+#include <AK/Types.h>
+
+namespace Kernel {
+
+enum class JailFlags : u32 {
+    None = 0,
+    NoPIDIsolation = 1 << 0,
+};
```

Comment:
In terms of design; conceptually jails could have a number of isolation features in the future that arguably can all be turned on or off. Instead of using the negative form here for flags ("no PID isolation") and specific APIs (`create_without_process_list`), how about using the positive approach by passing in `JailIsolationFlags` such as `::PIDIsolation` and only performing the isolation checks if a certain check is set? You could then default to a special flag by default that sets all isolation features.

This feels more "natural" to me than individually disabling features.

Reply:
While I agree it feels more natural to some users, I never designed the jail code so you would disable many features at once. I still need to look how it would look with this because this was not the original design in my mind (but it could be the "new" design if it's neat enough!).